### PR TITLE
[core] Use certificates from certifi if installed

### PR DESCRIPTION
**Pull request by coletdjnz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Come up a few times that using `certifi` for certificates may help solve some of the SSL issues we see (e.g. systems having no or outdated certificates). 

I'm not very familiar with this area so feedback or review is appreciated :smiley: 

Also suggestions for wording of docs, what I have is very rough and probably too technical explanation.

Added `no-certifi` compat-opt if there is a need to use system certificates (self signed?).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,6 +269,7 @@ While all the other dependencies are optional, `ffmpeg` and `ffprobe` are highly
 * [**secretstorage**](https://github.com/mitya57/secretstorage) - For accessing the Gnome keyring while decrypting cookies of Chromium-based browsers on Linux. Licensed under [BSD](https://github.com/mitya57/secretstorage/blob/master/LICENSE)
 * [**AtomicParsley**](https://github.com/wez/atomicparsley) - For embedding thumbnail in mp4/m4a if mutagen/ffmpeg cannot. Licensed under [GPLv2+](https://github.com/wez/atomicparsley/blob/master/COPYING)
 * [**brotli**](https://github.com/google/brotli) or [**brotlicffi**](https://github.com/python-hyper/brotlicffi) - [Brotli](https://en.wikipedia.org/wiki/Brotli) content encoding support. Both licensed under MIT <sup>[1](https://github.com/google/brotli/blob/master/LICENSE) [2](https://github.com/python-hyper/brotlicffi/blob/master/LICENSE) </sup>
+* [**certifi**](https://github.com/certifi/python-certifi) Provides Mozilla's root certificate bundle [License](https://github.com/certifi/python-certifi/blob/master/LICENSE)
```
Comment (pukkandan) [inline README.md:272]:
```suggestion
* [**certifi**](https://github.com/certifi/python-certifi) Provides Mozilla's root certificate bundle. Licensed under [Mozilla Public License](https://github.com/certifi/python-certifi/blob/master/LICENSE)
```

File: yt_dlp/utils.py [hunk 2]
```diff
@@ -85,6 +85,13 @@
     sockssocket,
 )
 
+try:
+    import certifi
+    has_certifi = True
+except ImportError:
+    certifi = None
```
Comment (pukkandan) [inline yt_dlp/utils.py:92]:
```suggestion
```

**Review by pukkandan — COMMENTED** [review]
Also, in https://github.com/coletdjnz/yt-dlp-dev/blob/db02f20007122411894a2a11595d56da072a145c/README.md#L281,
```diff
-The Windows and MacOS standalone release binaries are already built with the python interpreter, mutagen, pycryptodomex and websockets included.
+The Windows and MacOS standalone release binaries are already built with the python interpreter, and all optional python packages included.
```
since the list is becoming large

**Comment by gamer191** [comment]
> Also, in https://github.com/coletdjnz/yt-dlp-dev/blob/db02f20007122411894a2a11595d56da072a145c/README.md?plain=1#L281,
> 
> ```diff
> -The Windows and MacOS standalone release binaries are already built with the python interpreter, mutagen, pycryptodomex and websockets included.
> +The Windows and MacOS standalone release binaries are already built with the python interpreter and all optional python packages included.
> ```
> 
> since the list is becoming large

The average user won't know which dependencies are python packages. How about instead using a legend, such as putting a * next to all the dependencies which are included in the binaries?

File: README.md [hunk 3]
```diff
@@ -268,6 +269,7 @@ While all the other dependencies are optional, `ffmpeg` and `ffprobe` are highly
 * [**secretstorage**](https://github.com/mitya57/secretstorage) - For accessing the Gnome keyring while decrypting cookies of Chromium-based browsers on Linux. Licensed under [BSD](https://github.com/mitya57/secretstorage/blob/master/LICENSE)
 * [**AtomicParsley**](https://github.com/wez/atomicparsley) - For embedding thumbnail in mp4/m4a if mutagen/ffmpeg cannot. Licensed under [GPLv2+](https://github.com/wez/atomicparsley/blob/master/COPYING)
 * [**brotli**](https://github.com/google/brotli) or [**brotlicffi**](https://github.com/python-hyper/brotlicffi) - [Brotli](https://en.wikipedia.org/wiki/Brotli) content encoding support. Both licensed under MIT <sup>[1](https://github.com/google/brotli/blob/master/LICENSE) [2](https://github.com/python-hyper/brotlicffi/blob/master/LICENSE) </sup>
+* [**certifi**](https://github.com/certifi/python-certifi) Provides Mozilla's root certificate bundle. Licensed under [Mozilla Public License](https://github.com/certifi/python-certifi/blob/master/LICENSE)
```
Comment (gamer191) [inline README.md:272]:
```suggestion
* [**certifi**](https://github.com/certifi/python-certifi) Provides Mozilla's root certificate bundle. Licensed under [Mozilla Public License v2](https://github.com/certifi/python-certifi/blob/master/LICENSE)
```
Excuse my pedanticism

**Pull request merged**